src/praisonai-agents/praisonaiagents/agent/chat_mixin.py

```
@@ -1876,7 +1876,7 @@ def _apply_context_management(
             logging.warning(f"Context management error (continuing without): {e}")
             return messages, None
 
-    def _truncate_tool_output(self, tool_name: str, output: str) -> str:
+    def _truncate_tool_output(self, tool_name: str, output: str, tool_call_id: str = None) -> str:
```

Ruff (0.15.17)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)